flow/Stats.actor.cpp

```diff
@@ -45,21 +45,32 @@ double Counter::getRate() const {
 }
 
 double Counter::getRoughness() const {
-	double elapsed = now() - interval_start;
+	double elapsed = now() - roughness_interval_start;
 	if(elapsed == 0) {
 		return 0;
 	}
 
+	// If we have time samples t in T, and let:
+	// n = size(T) = interval_delta
+	// m = mean(T) = elapsed / interval_delta
+	// v = sum(t^2) for t in T = interval_sq_time
+	// 
+	// The formula below is: (v/(m*n)) / m - 1
+	// This is equivalent to (v/n - m^2) / m^2 = Variance(T)/m^2
+	// Variance(T)/m^2 is equal to Variance(t/m) for t in T
 	double delay = interval_sq_time / elapsed;
-	return delay * getRate() * 2;
+	return delay * interval_delta / elapsed - 1;
```

Comment:
I suppose this will give a roughness of -1 if there is no data for a window. Not sure if that's desirable or not.

Reply:
It's not substantially different from before, where you would get a value of 0 in this case, which is below the otherwise minimum value of 1. Line 50 also returns 0, maybe it should be updated to return -1 too?

Reply:
> maybe it should be updated to return -1 too?

I think so. A variance of 0 still has a valid meaning: no variance. -1 is definitely an indicator of invalid value.